draft-ietf-curdle-gss-keyex-sha2.xml

```diff
@@ -176,201 +176,64 @@
         information on GSS-API context establishment operations, Section 3
         being the most relevant. All Security Considerations described in
-        <xref target="RFC7546"/> apply here too.</t>
-
-        <t>The Client:</t>
-        <t>1. C generates an ephemeral key pair with public key Q_C. It does
-            that by:
-          <list>
-            <t>For NIST curves:
-              <list>
-                <t>Following the algorithm in Section 3.2.1 of
-                    <xref target="SEC1v2"/> to create a private key d_C and
-                    public key q_C using the curve paramaters associated
-                    with the selected key exchange method.</t>
-                <t>Converting the resulting point q_C to the Q_C octet string
-                    using the algorithm in Section 2.3.3 of
-                    <xref target="SEC1v2"/> to create uncompressed point
-                    representation.</t>
-              </list>
-            </t>
-            <t>For curve25519 and curve448:
-                <list>
-                    <t>Following the algorithm in Section 6.1 of
-                        <xref target="RFC7748"/>, the private key d_C and
-                        the base point g is prepared.</t>
-                    <t>Calculating Q_C as the result of the call to X25519
-                        or X448 function, respectively for curve25519 and
-                        curve448 key exchange, with parameters d_C and g.</t>
-                </list>
-            </t>
-          </list>
-        </t>
-        <t>2. C calls GSS_Init_sec_context(), using the most recent reply
-        token received from S during this exchange, if any. For this call,
-        the client MUST set mutual_req_flag to "true" to request that mutual
-        authentication be performed. It also MUST set integ_req_flag to "true"
-        to request that per-message integrity protection be supported for this
-        context. In addition, deleg_req_flag MAY be set to "true" to request
-        access delegation, if requested by the user. Since the key exchange
-        process authenticates only the host, the setting of anon_req_flag is
-        immaterial to this process. If the client does not support the
-        "gssapi-keyex" user authentication method described in Section 4 of
+        <xref target="RFC7546"/> apply here too.
+
+        <t>The parties generate each an ephemeral key pair, according to
+        Section 3.2.1 of <xref target="SEC1v2"/>. Keys are verified upon
+        receipt by the parties according to Section 3.2.3.1 of
+        <xref target="SEC1v2"/>.
+
+        <t>For NIST Curves keys use uncompressed point representation and
+        must be converted using the algorithm in Section 2.3.4 of
+        <xref target="SEC1v2"/>. If the conversion fails or the point is
+        trasmitted using compressed representation, the key exchange MUST
+        fail.</t>
+
```

Comment:
Description of "K" missing.

```
<t>For NIST curves the shared secret K is the result of conversion the shared secret
field element received as output from the ECDH primitive from Section 3.3.1 of 
<xref target="SEC1v2"/> to an octet string using the field element to octet string
conversion from section 2.3.5 of <xref target="SEC1v2"/>.</t>

<t>For curve25519 and curve448, the shared key K is the output of the call to
X25519 and X448 as is.</t>
```

Reply:
I must have accidentally removed the section where I say the parties perform ECDH share dsecret computation., fixing.

Reply:
yes, it was intertwined with the descriptions so it went away with them

this is the new proposed description